README.md

```diff
@@ -1,264 +1,483 @@
+environment variables ONLY) in the app instead of environment variables, see the
+[Authorization Examples](#authorization-examples) and/or [system properties](#system-properties)
+section below.
+
+The following environment variables need to be included in the apps runtime environment in
+order use the Conjur API:
+
+- `CONJUR_APPLIANCE_URL` - The URL of the Conjur instance you are connecting to. When connecting to
+  DAP configured for high availability, this should be the URL of the master load balancer (if
+  performing read and write operations) or the URL of a follower load balancer (if performing
+  read-only operations).
+- `CONJUR_ACCOUNT` - Conjur account that you are connecting to. This value is set during Conjur deployment.
+- `CONJUR_AUTHN_LOGIN` - User/host identity
+- `CONJUR_AUTHN_API_KEY` - User/host API key
+- `CONJUR_AUTHN_URL` - (optional) Alternate authentication endpoint. By default the client
+  uses the standard `<applianceUrl>/authn` for generic username and API key login flow.
+
+For example, you can specify the environment variables like so:
+
+```sh-session
+export CONJUR_APPLIANCE_URL=https://conjur.myorg.com/api
+export CONJUR_ACCOUNT=myorg
+export CONJUR_AUTHN_LOGIN=host/myhost.example.com
+export CONJUR_AUTHN_API_KEY=sb0ncv1yj9c4w2e9pb1a2s
+```
 
-`CONJUR_AUTHN_URL` - (optional) Set this variable if using alternative authentication method
+or you could provide these at runtime to your jar:
+```sh-session
+$ CONJUR_APPLIANCE_URL=https://conjur.myorg.com/api \
+  CONJUR_ACCOUNT=myorg \
+  CONJUR_AUTHN_LOGIN=host/myhost.example.com \
+  CONJUR_AUTHN_API_KEY=sb0ncv1yj9c4w2e9pb1a2s \
+  java -jar myConjurClient.jar
+```
+If you are using a host-based user like this example shows, you will need to add the host
+to Conjur with the proper privileges in policy in order to know the appropriate
+`CONJUR_AUTHN_LOGIN` and `CONJUR_AUTHN_API_KEY` values.
+
+### System Properties
+
+This API can also be configured using [Java system properties](https://www.baeldung.com/java-system-get-property-vs-system-getenv).
+You can specify any portion (or all) of the configuration values this way. The advantage
+of this approach is that the values can be changed dynamically as needed. For example,
+this snippet would let your client be able to use the API methods using properties defined
+from the CLI:
+```sh-session
+java -jar myConjurClient.jar \
+     -DCONJUR_APPLIANCE_URL=https://conjur.myorg.com/api \
+     -DCONJUR_ACCOUNT=myorg \
+     -DCONJUR_AUTHN_LOGIN=host/myhost.example.com \
+     -DCONJUR_AUTHN_API_KEY=sb0ncv1yj9c4w2e9pb1a2s
+```
 
-For example, specify the environment variables like so:
+If you are using Maven, you can also specify these proprerties on the CLI:
 
-```bash
-CONJUR_ACCOUNT=myorg
-CONJUR_AUTHN_LOGIN=host/myhost.example.com
-CONJUR_AUTHN_API_KEY=sb0ncv1yj9c4w2e9pb1a2s
-CONJUR_APPLIANCE_URL=https://conjur.myorg.com/api
+```sh-session
+mvn exec:java \
```

Comment:
If they aren't sensitive, it's probably move common to provide them in the `pom.xml` file itself within `<properties> element.  See [docs](https://maven.apache.org/pom.html#Properties).

Reply:
True but I didn't have time to investigate if those persist in the created artifact. That part may be for a followup issue.

Reply:
If they are put into the parent project `pom.xml`, they should apply to included compiled, dependencies.  But, still, fine to leave for now.

Reply:
Fair enough - it's definitely something that could be expanded here (once we have a bit more time on it)